homework_02/victor_dimitrov_04/views.py

```diff
@@ -0,0 +1,49 @@
+from django.shortcuts import render
+import pytz
+import datetime
+
+# Create your views here.
+
+import requests
+
+def trains(request):
+
+    initial_filters = {
+        'page[offset]': '0',
+        'page[limit]': '50',
+        'include': 'trip,schedule',
+        'filter[stop]':'place-north',
+        'filter[route_type]': '2',
+    }
+
+    response = requests.get('https://api-v3.mbta.com/predictions', params=initial_filters).json()
+
+    trains, trip_ids = [], []
+
+    for element in response['data']:
+        train_info = {}
+        train_info['status'] = element['attributes']['status']
+        train_info['route'] = element['relationships']['route']['data']['id'][3:]
+
+        #I am lazy person - the numbers on those indices attended both in the route id and the track id
+        #That indicated those numbers were connected to something, which after some search appeared to be
+        #the train/track/vehicle number. 
+        trip_id = element['relationships']['trip']['data']['id']
+        trip_ids.append(trip_id)
+        train_info['number'] = trip_id[10:14]
```

Comment:
By using indexes, you limit the results. Imagine that we have more than 50. It's not a big deal, it works for this homework.